model/canvas.cc

```diff
@@ -145,9 +145,25 @@ namespace minsky
       }
     if (fromPort.get())
       {
-          if (auto to=closestInPort(x,y))
+          if (auto to=closestInPort(x,y)) {
             model->addWire(static_cast<shared_ptr<Port>&>(fromPort),to);
-        fromPort.reset();
+            fromPort.reset();
+            auto toItem = to.get()->item();
+            if (auto ravel=static_cast<Ravel&>(toItem))
```

Comment:
This won't work. static_cast<Ravel&> just tells the compiler that this is a Ravel, regardless of whether it actually is.
In fact it is not, currently, because it has been sliced on line 151.

What you want is `auto ravel=dynamic_cast<Ravel*>(&toItem)`, as dynamic_cast returns nullptr if it is not, in fact a Ravel.

dynamic_cast<Ravel&> is different - it throw an exception it the item is not a Ravel - sometimes that is what you want to do - no checks are required before using the result.